src/views/operations/useGetOperations.tsx

```diff
@@ -18,103 +19,102 @@ import {
 
 const REFRESH_INTERVAL = 15000;
 
-// TODO: Add tests
-export const useGetOperations = (initialAddresses: RawPkh[]) => {
+type QueryParams =
+  | {
+      lastId?: number;
+      limit?: number;
+      sort?: "asc" | "desc";
+    }
+  | undefined;
+
+/**
+ * Hook to fetch operations for given addresses.
+ *
+ * It fetches the latest operations on the first load and allows
+ * to load more (older) operations as needed.
+ *
+ * Every {@link REFRESH_INTERVAL} it fetches the latest operations
+ * and prepends it to the already fetched operations list.
+ *
+ * If the refresh button is clicked then it will trigger this behaviour immediately.
+ *
+ * @param addresses - list of addresses to fetch operations related to
+ * @returns
+ *   operations - operations ordered by id in descending order
+ *   isFirstLoad - true if the first load is in progress
+ *   isLoading - true if the next page is being fetched
+ *   hasMore - true if there are more operations to fetch
+ *   loadMore - function to load more operations (older ones)
+ */
+export const useGetOperations = (addresses: RawPkh[]) => {
   const network = useSelectedNetwork();
-  const [operations, setOperations] = useState<TzktCombinedOperation[]>([]);
-  const [hasMore, setHasMore] = useState(true);
-  const [isFirstLoad, setIsFirstLoad] = useState(true);
-  const { isLoading, handleAsyncAction } = useAsyncActionHandler();
-
-  const [addresses, setAddresses] = useState<RawPkh[]>(initialAddresses);
   const dispatch = useAppDispatch();
+  const refetchTrigger = useRefetchTrigger();
 
-  const [updatesTrigger, setUpdatesTrigger] = useState(0);
+  const {
+    isFetching,
+    data: operations,
+    hasNextPage,
+    isLoading,
+    fetchNextPage,
+    fetchPreviousPage,
+  } = useInfiniteQuery({
+    queryKey: ["operations", addresses, dispatch, network],
+    initialPageParam: undefined as QueryParams,
+    queryFn: async ({ pageParam }) =>
+      fetchOperationsAndUpdateTokensInfo(dispatch, network, addresses, pageParam),
+    getNextPageParam: lastPage => {
+      if (lastPage.length === 0) {
+        return undefined;
+      }
+
+      const lastId = lastPage[lastPage.length - 1].id;
+      return { lastId };
+    },
+
+    getPreviousPageParam: (_, pages) => {
+      const lastId = maxBy(pages.flat(), "id")?.id;
+
+      // if lastId is not defined it means that there are no operations yet for the accounts
+      // but we still need to keep updating the operations list to get ones once they appear
+      return lastId ? { lastId, sort: "asc" as const } : undefined;
```

Comment:
it didn't. it is confusing I agree, but previous page here means a page with newer records. since we sort desc by id, it's much more convenient to *prepend* new records to the beginning of the pages array.

and the `sort: "asc"` param here makes sure that we fetch what comes after the current biggest id

I hope it makes sense, please let me know otherwise

Reply:
I see, makes sense now

`previous page here means a page with newer records` - this took me some time to figure it out, so maybe we can have a comment clarifying what next and prev pages mean?

My understanding is that nextPage - fetches older records, until we still have some
And prevPage getting the newer records that were added at the top after first page was fetched
Is that correct?

Reply:
That's correct